gdal/frmts/netcdf/netcdfdataset.cpp

```diff
@@ -4849,48 +4868,50 @@ double netCDFDataset::rint( double dfX )
 CPLErr netCDFDataset::ReadAttributes( int cdfidIn, int var)
 
 {
-    char szVarName[NC_MAX_NAME + 1];
-    int nbAttr = 0;
+    char *pszVarFullName = nullptr;
+    ERR_RET(NCDFGetVarFullName(cdfidIn, var, &pszVarFullName));
+    size_t nMetaNameSize = sizeof(char) * (strlen(pszVarFullName) + 1
+                                           + NC_MAX_NAME + 1);
+    char *pszMetaName = (char *) CPLMalloc(nMetaNameSize);
 
-    int status = nc_inq_varnatts(cdfidIn, var, &nbAttr);
-    NCDF_ERR(status);
-    if( status != NC_NOERR )
-        return CE_Failure;
-    if( var == NC_GLOBAL )
-    {
-        strcpy(szVarName, "NC_GLOBAL");
-    }
-    else
-    {
-        szVarName[0] = '\0';
-        status = nc_inq_varname(cdfidIn, var, szVarName);
-        NCDF_ERR(status);
-        if( status != NC_NOERR )
-            return CE_Failure;
-    }
+    int nbAttr = 0;
+    NCDF_ERR(nc_inq_varnatts(cdfidIn, var, &nbAttr));
 
     for( int l = 0; l < nbAttr; l++ )
     {
         char szAttrName[NC_MAX_NAME + 1];
         szAttrName[0] = 0;
-        status = nc_inq_attname(cdfidIn, var, l, szAttrName);
-        NCDF_ERR(status);
-        char szMetaName[NC_MAX_NAME * 2 + 1 + 1];
-        snprintf(szMetaName, sizeof(szMetaName), "%s#%s", szVarName,
-                 szAttrName);
+        NCDF_ERR(nc_inq_attname(cdfidIn, var, l, szAttrName));
+        snprintf(pszMetaName, nMetaNameSize,
+                 "%s#%s", pszVarFullName, szAttrName);
 
         char *pszMetaTemp = nullptr;
         if( NCDFGetAttr(cdfidIn, var, szAttrName, &pszMetaTemp) == CE_None )
         {
             papszMetadata =
-                CSLSetNameValue(papszMetadata, szMetaName, pszMetaTemp);
+                CSLSetNameValue(papszMetadata, pszMetaName, pszMetaTemp);
             CPLFree(pszMetaTemp);
             pszMetaTemp = nullptr;
         }
         else
         {
-            CPLDebug("GDAL_netCDF", "invalid global metadata %s", szMetaName);
+            CPLDebug("GDAL_netCDF", "invalid metadata %s", pszMetaName);
+        }
+    }
+
+    CPLFree(pszVarFullName);
+    CPLFree(pszMetaName);
+
+    if( var == NC_GLOBAL )
+    {
+        // Recurse on sub-groups.
+        int nSubGroups = 0, *panSubGroupIds = nullptr;
```

Comment:
separate declarations on 2 lines

Reply:
Are multiple declarations on one line forbidden? Or is it because we cannot mix value and pointer declaration?
Is there a GDAL coding convention specifying these kind of things? I only found https://trac.osgeo.org/gdal/wiki/rfc8_devguide

Reply:
For both reasons. This is probably not specified in the devguide, but this is something I try to stick to personally for clarity